connector_prestashop/tests/test_import_sale.py

```diff
@@ -157,15 +158,25 @@ def test_import_sale_record(self):
 
         # import of the sale order
         with recorder.use_cassette('test_import_sale_record_5'):
-            result = self.env['prestashop.sale.order'].import_record(
-                self.backend_record, 5)
 
-        error_msg = ('Import of the order 5 canceled '
-                     'because it has not been paid since 30 days')
-        self.assertEqual(result, error_msg)
+            with self.backend_record.work_on(
+                    model_name='prestashop.sale.order') as work:
+                order = work.component(usage='backend.adapter').read(5)
 
-        with recorder.use_cassette('test_import_sale_record_5'):
-            with freeze_time("2016-12-08"):
+            fmt = '%Y-%m-%d %H:%M:%S'
+            order_date = datetime.strptime(order['date_add'], fmt)
+            order_date_29 = (order_date + timedelta(days=29)).strftime(fmt)
+            order_date_31 = (order_date + timedelta(days=31)).strftime(fmt)
+
+            with freeze_time(order_date_31):
+                result = self.env['prestashop.sale.order'].import_record(
+                    self.backend_record, 5)
+
+            error_msg = ('Import of the order 5 canceled '
+                         'because it has not been paid since 30 days')
+            self.assertEqual(result, error_msg)
+
+            with freeze_time(order_date_29):
```

Comment:
```
ERROR: test_import_sale_record (odoo.addons.connector_prestashop.tests.test_import_sale.TestImportSale)
` Import a sale order
Traceback (most recent call last):
`   File "connector_prestashop/tests/common.py", line 71, in _decorated
`     result = func(self, *args, **kwargs)
`   File "connector_prestashop/tests/test_import_sale.py", line 170, in test_import_sale_record
`     self.backend_record, 5)
`   File "connector_prestashop/models/binding/common.py", line 47, in import_record
`     return importer.run(prestashop_id, force=force)
`   File "connector_prestashop/components/importer.py", line 272, in run
`     self._import(binding, **kwargs)
`   File "connector_prestashop/components/importer.py", line 287, in _import
`     record = self._create_data(map_record)
`   File "connector_prestashop/components/importer.py", line 125, in _create_data
`     return map_record.values(for_create=True)
`   File "connector/components/mapper.py", line 979, in values
`     values = self._mapper._apply(self, options=options)
`   File "connector/components/mapper.py", line 753, in _apply
`     return self._apply_with_options(map_record)
`   File "connector/components/mapper.py", line 802, in _apply_with_options
`     to_attr, model_name)
`   File "connector_prestashop/models/sale_order/importer.py", line 194, in _map_child
`     detail_record = adapter.read(child_record['id'])
`   File "connector_prestashop/components/backend_adapter.py", line 197, in read
`     res = self.client.get(self._prestashop_model, id, options=attributes)
`   File "/home/paco/workspace/ptic-saas-server-10/eggs/prestapyt-0.9.0-py2.7.egg/prestapyt/prestapyt.py", line 393, in get
`     return self.get_with_url(full_url)
`   File "/home/paco/workspace/ptic-saas-server-10/eggs/prestapyt-0.9.0-py2.7.egg/prestapyt/prestapyt.py", line 566, in get_with_url
`     response = super(PrestaShopWebServiceDict, self).get_with_url(url)
`   File "/home/paco/workspace/ptic-saas-server-10/eggs/prestapyt-0.9.0-py2.7.egg/prestapyt/prestapyt.py", line 401, in get_with_url
`     return self._parse(self._execute(url, 'GET').content)
`   File "/home/paco/workspace/ptic-saas-server-10/eggs/prestapyt-0.9.0-py2.7.egg/prestapyt/prestapyt.py", line 241, in _execute
`     headers=request_headers,
`   File "/home/paco/workspace/ptic-saas-server-10/virtualenv/lib/python2.7/site-packages/requests/sessions.py", line 508, in request
`     resp = self.send(prep, **send_kwargs)
`   File "/home/paco/workspace/ptic-saas-server-10/virtualenv/lib/python2.7/site-packages/requests/sessions.py", line 618, in send
`     r = adapter.send(request, **kwargs)
`   File "/home/paco/workspace/ptic-saas-server-10/virtualenv/lib/python2.7/site-packages/requests/adapters.py", line 440, in send
`     timeout=timeout
`   File "/home/paco/workspace/ptic-saas-server-10/virtualenv/lib/python2.7/site-packages/urllib3/connectionpool.py", line 601, in urlopen
`     chunked=chunked)
`   File "/home/paco/workspace/ptic-saas-server-10/virtualenv/lib/python2.7/site-packages/urllib3/connectionpool.py", line 380, in _make_request
`     httplib_response = conn.getresponse(buffering=True)
`   File "/home/paco/workspace/ptic-saas-server-10/eggs/vcrpy-1.10.5-py2.7.egg/vcr/stubs/__init__.py", line 232, in getresponse
`     self.cassette.record_mode)
` CannotOverwriteExistingCassetteException: No match for the request (<Request (GET) http://localhost:8080/api/order_details/13>) was found. Can't overwrite existing cassette ('connector_prestashop/tests/fixtures/cassettes/test_import_sale_record_5.yaml') in your current record mode ('once').
```

In a fresh PS orders are not older than 30 days. We have fixed this by querying the date of the order and freezing the time after and before 30 days from the date.